File: Data Structures & Algorithms/duplicate-integer/submission-2.cpp
class Solution {
public:
    bool hasDuplicate(vector<int>& nums) {
        unordered_map<int,int> map;
        for(int x : nums){
            map[x]++;
        }
        for(const auto& [num,count] : map){
            if(count > 1) return true;

        }
        return false;
    }
};

File: Data Structures & Algorithms/top-k-elements-in-list/submission-2.py
import heapq
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        heap = []
        freq = {}
        for x in nums:
            freq[x] = freq.get(x,0)+1
        for num in freq.keys():
            heapq.heappush(heap,(freq[num],num))
            if len(heap) > k:
                heapq.heappop(heap)
        result = []
        for i in range(k):
            result.append(heapq.heappop(heap)[1])
        return result